lib/sport_ngin_aws_auditor/scripts/audit.rb

```diff
@@ -47,76 +48,79 @@ def self.gather_data(class_type, tag_name)
           instances_without_tag = klass.filter_instance_without_tags(instances)
           instance_hash = klass.instance_count_hash(instances_without_tag)
           klass.add_instances_with_tag_to_hash(instances_with_tag, instance_hash)
-          return instance_hash
         elsif options[:reserved]
-          return klass.instance_count_hash(klass.get_reserved_instances)
+          instance_hash = klass.instance_count_hash(klass.get_reserved_instances)
         else
-          return klass.compare(tag_name)
+          instance_hash = klass.compare(tag_name)
         end
+
+        compared_array = []
+        instance_hash.each do |key, value|
```

Comment:
We can improve this even more in the future by encapsulating the list of objects returned to this instance_hash as a class, but I think that your changes here are a great step in that direction and gives us some colors. 👍